weave_query/weave_query/io_service.py

```diff
@@ -481,12 +502,26 @@ def response_task_ended(self, task: asyncio.Task) -> None:
 
     async def close(self) -> None:
         self.connected = False
+        sentinel_response = ServerResponse(http_error_code=200, error=False, client_id=self.client_id, id=SENTINEL_ID, value=None)
+        await self.response_queue.async_put(sentinel_response)
+        await self.response_task
```

Comment:
AFAIK this is not used anywhere right now, all of our clients are sync. Is there a motivation to switch to AsyncConnections?

Reply:
I saw this being used here: https://github.com/wandb/weave/blob/6a093a4f2d8088e0d5e02c6ca2a5688cc017ac0d/weave_query/weave_query/ops_domain/table.py#L773

I didn't see an example of op code making http requests, so I tried to implement something similar based on that use of io_service.